cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletEndpointTest.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet;
+
+import com.linkedin.kafka.cruisecontrol.async.OperationFuture;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.UserTasksParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.CruiseControlResponse;
+import com.linkedin.kafka.cruisecontrol.servlet.response.UserTaskState;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import kafka.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class KafkaCruiseControlServletEndpointTest {
+  private static class MockResult implements CruiseControlResponse {
+    public void discardIrrelevantResponse(CruiseControlParameters parameters) { }
+    public void writeSuccessResponse(CruiseControlParameters parameters, HttpServletResponse response) throws
+                                                                                                       IOException {
+    }
+  }
+
+  @Test
+  public void testUserTaskParameters() throws Exception, UnsupportedEncodingException {
```

Comment:
Would it be possible to:
1. Use smaller methods to split the monolithic code to smaller methods.
2. Move constants to relevant `static` variables.
3. Consider using `Parameterized` testing -- e.g. see `RandomGoalTest`.